src/collections/queue/README.md

```diff
@@ -1,3 +1,74 @@
 # 佇列 Queue
 
+![Queue - Wiki](https://upload.wikimedia.org/wikipedia/commons/thumb/5/52/Data_Queue.svg/450px-Data_Queue.svg.png)
+
+佇列是一個具有*先進先出* FIFO 特性的資料結構。從 Wikipedia 上的圖為例，一個資料從左邊進入佇列並從右邊離開，最先進入佇列的資料會最先被取出。
+
+佇列常見實作方式有：陣列 array、鏈結串列 linked list、堆疊 stack。為了使概念容易理解，我們選擇以類似陣列的 Vector 實作。
+
+## 基本操作
+
+- `enqueue`：將新資料加入佇列
+- `dequeue`：將最先放入的資料移出佇列
+- `peek`：在不將資料移出佇列的情況下取得最先放入的資料
+- `size`：取得佇列大小
+
+## 架構設計
+
+```rust
+pub struct Queue<T> {
+    items: Vec<T>,
+}
+```
+
+以 `items` 保存加入佇列的資料。再來我們將 `items[0]` 當作 rear，並將 `items[length-1]` 當作 front，所以並不需要定義 front 和 rear 在 struct 裡面。
```

Comment:
What do front and rear represent here? More explanation may be better?

Also, `items[0]` `items[length-1]` seems to be a specific programming language syntax, it would be better to elaborate this idea with plain Chinese.